internal/secrets/secrets.go

```diff
@@ -0,0 +1,219 @@
+package secrets
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/alecthomas/kingpin/v2"
```

Comment:
dependency on kingpin should be avoided as much as possible from low-level packages like this, kingpin is really only meant to be used from cli and testing packages